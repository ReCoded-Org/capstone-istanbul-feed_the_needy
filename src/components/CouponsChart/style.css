.couponsChart {
  padding: 1rem 0;
}

.couponsChartTitle {
  padding: 1rem 0;
}

.couponsChartStatistics {
  padding-bottom: 1rem;
}

.spin {
  padding-top: inherit;
  color: #e17007;
  width: 100%;
}

.spin .ant-spin-dot-item {
  background-color: #e17007;
}
