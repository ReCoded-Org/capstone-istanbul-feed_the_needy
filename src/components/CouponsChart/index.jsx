import React, { useEffect, useState } from "react";
import "./style.css";
import { Bar } from "react-chartjs-2";
import { Typography, Divider, Statistic, Row, Col, Card, Spin } from "antd";
import { useTranslation } from "react-i18next";
import firebase from "../../firebaseConfig";

const db = firebase.firestore();
const auth = firebase.auth();
const { Title } = Typography;

const CouponsChart = () => {
  const { t } = useTranslation();
  const [soldCoupons, setSoldCoupons] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (user) {
        const unsubscribe = db
          .collection("soldCoupons")
          .doc(user.uid)
          .onSnapshot((snapshot) => {
            const dataArr = [];
            dataArr.push({ ...snapshot.data() });
            setSoldCoupons(dataArr[0]);
            setLoading(false);
          });
        return unsubscribe;
      }
      return user;
    });
  }, []);

  const labelList = t("couponsChart.labels", { returnObjects: true });
  const labels = ["25₺", "50₺", "75₺", "100₺", "150₺", "200₺"];
  const bgColors = ["purple", "gold", "blue", "orange", "red", "green"];
  const dataList = [
    soldCoupons["25₺"],
    soldCoupons["50₺"],
    soldCoupons["75₺"],
    soldCoupons["100₺"],
    soldCoupons["150₺"],
    soldCoupons["200₺"],
  ];
  const datasetList = [];

  for (let i = 0; i < labels.length; i += 1) {
    datasetList.push({
      label: labels[i],
      bgColor: bgColors[i],
      data: dataList[i],
    });
  }
  const data = {
    labels: labelList,
    datasets: datasetList.map((x) => ({
      label: x.label,
      backgroundColor: x.bgColor,
      data: x.data,
    })),
  };

  const statisticsData = [];
  const values = [10, 50, 500, 1000];
  const titles = t("couponsChart.statistics", { returnObjects: true });

  for (let i = 0; i < values.length; i += 1) {
    statisticsData.push({ title: titles[i], value: values[i] });
  }

  return (
    <div className="couponsChart">
      <Divider className="couponsChartTitle">
        <Title level={3}>{t("couponsChart.title")}</Title>
      </Divider>
      {loading ? (
        <Spin className="spin" size="large" tip={t("spin")} />
      ) : (
        <>
          <Row className="couponsChartStatistics" gutter={[16, 16]}>
            {statisticsData.map((stat) => (
              <Col key={stat.title} xs={24} sm={12} md={12} lg={6} xl={6}>
                <Card hoverable>
                  <Statistic title={stat.title} value={stat.value} />
                </Card>
              </Col>
            ))}
          </Row>
          <Bar
            data={data}
            width={window.screen.width < 576 ? 1500 : 1000}
            height={window.screen.width < 576 ? 1000 : 250}
            options={{
              maintainAspectRatio: true,
            }}
          />
        </>
      )}
    </div>
  );
};

export default CouponsChart;
