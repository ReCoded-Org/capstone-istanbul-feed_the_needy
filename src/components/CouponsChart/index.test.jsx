import React from "react";
import renderer from "react-test-renderer";
import CouponsChart from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("CouponsChart renders correctly", () => {
  const tree = renderer.create(<CouponsChart />).toJSON();
  expect(tree).toMatchSnapshot();
});
