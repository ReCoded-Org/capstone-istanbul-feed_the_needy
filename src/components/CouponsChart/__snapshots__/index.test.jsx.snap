// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponsChart renders correctly 1`] = `
<div
  className="couponsChart"
>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center couponsChartTitle"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      <h3
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        couponsChart.title
      </h3>
    </span>
  </div>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text spin"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
    <div
      className="ant-spin-text"
    >
      spin
    </div>
  </div>
</div>
`;
